gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -65,6 +71,27 @@ public static void writeTokenToFile(Token<? extends TokenIdentifier> token, Path
     credentials.writeTokenStorageFile(tokenFilePath, configuration);
   }
 
+  /**
+   * Update {@link Token} with the configured token path.
+   *
+   * @param config a {@link Config} object carrying Hadoop configuration properties and token path config
+   * @throws IOException
+   */
+  public static void updateToken(Config config) throws IOException{
+    FileSystem fs = GobblinClusterUtils.buildFileSystem(config, new Configuration());
+    Path tokenPath = new Path(fs.getHomeDirectory(),
+        config.getString(GobblinYarnConfigurationKeys.APPLICATION_NAME_KEY) + Path.SEPARATOR
+            + GobblinYarnConfigurationKeys.TOKEN_FILE_NAME);
+    if(fs.exists(tokenPath)) {
+      Credentials credentials = Credentials.readTokenStorageFile(tokenPath, fs.getConf());
```

Comment:
Can lines 86-91 be refactored out into a separate addCredentials() method, which can be called also from YarnContainerSecurityManager?